### PR TITLE
Add en-CA translations.

### LINKED ISSUE #881: Add en-CA translations
# Feature Request

## Description
Add en-CA translations to all applicable components in terra-framework to support the Canadian language.

## Additional Context / Screenshots
String translations for en-CA should match en-US with the exception of the date format (if applicable to the component) that is all-numeric. For example, the all-numeric date format of `MM/DD/YYYY` in en-US should be `YYYY-MM-DD` in en-CA.  Any date format that is not all numeric should match en-US. See the [all-numeric dates](https://en.wikipedia.org/wiki/Date_and_time_notation_in_Canada) section or [date format for Canada](https://en.wikipedia.org/wiki/Date_format_by_country) for more details.

### PR DESCRIPTION
**Pull request by benbcai**
### Summary

- Fixes #881
- Add en-CA translations

### CODE REVIEW

File: packages/terra-application-layout/translations/en-CA.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "Terra.applicationLayout.applicationHeader.menuToggleLabel": "Toggle Menu",
+  "Terra.applicationLayout.utilityDefaults.menu": "Menu",
+  "Terra.applicationLayout.utilityDefaults.userInformation": "User Information",
+  "Terra.applicationLayout.utilityDefaults.changePhoto": "Change Photo",
+  "Terra.applicationLayout.utilityDefaults.settings": "Settings",
+  "Terra.applicationLayout.utilityDefaults.appearance": "Appearance",
+  "Terra.applicationLayout.utilityDefaults.security": "Security",
+  "Terra.applicationLayout.utilityDefaults.help": "Help",
+  "Terra.applicationLayout.utilityDefaults.gettingStarted": "Getting Started",
+  "Terra.applicationLayout.utilityDefaults.about": "About",
```
Comment (tbiethman) [inline packages/terra-application-layout/translations/en-CA.json]: Pretty sure this should be "Aboot"

**Comment by benbcai** [comment]
@emilyrohrbough Are you referring this PR: https://github.com/cerner/terra-aggregate-translations/pull/14? Once terra-aggregate-translations is released and gets pulled in, then I would think it should check for en-CA.

File: packages/terra-application-layout/translations/en-CA.json (hunk 1 above)
Comment (benbcai) [reply packages/terra-application-layout/translations/en-CA.json]: Are you serious? https://www.atlasobscura.com/articles/whats-going-on-with-the-way-canadians-say-about

**Comment by emilyrohrbough** [comment]
@benbcai  Yes it will pull the translations once its release, I wasn't sure when the release would happen, so I figured it'd be a good spot check so we can merge the PR without depending on the latest release. I don't think we need to merge the config changes in

File: packages/terra-application-layout/translations/en-CA.json (hunk 1 above)
Comment (tbiethman) [reply packages/terra-application-layout/translations/en-CA.json]: most definitely not serious

**Comment by benbcai** [comment]
The build currently shows that these components are missing en-CA translations when looking in the node_modules folder. I think terra-dev-site needs to be released in order for the node-modules to be updated. I did verify manually that all of these components have en-CA translations.

```
[terra-aggregate-translations] Aggregating translations for en,en-AU,en-CA,en-US,en-GB,es,es-US,es-ES,de,fr,fr-FR,nl,nl-BE,pt,pt-BR,sv,sv-SE locales.
There was an error reading your translations file /opt/module/node_modules/terra-abstract-modal/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-abstract-modal/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-action-header/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-action-header/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-application-navigation/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-application-navigation/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-dev-site/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-dev-site/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-form-checkbox/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-form-checkbox/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-form-field/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-form-field/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-form-radio/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-form-radio/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-form-select/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-form-select/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-i18n/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-i18n/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-icon/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-icon/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-infinite-list/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-infinite-list/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-menu/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-menu/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-notification-dialog/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-notification-dialog/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-overlay/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-overlay/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-popup/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-popup/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-search-field/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-search-field/translations/en-CA.json' 
There was an error reading your translations file /opt/module/node_modules/terra-status-view/translations/en-CA.json.
 Exception Message: ENOENT: no such file or directory, open '/opt/module/node_modules/terra-status-view/translations/en-CA.json' 
```

**Pull request merged**